lib/fluent/output.rb

```diff
@@ -557,16 +557,20 @@ def emit(tag, es, chain)
       @emit_count += 1
       formatted_data = {}
       es.each {|time,record|
-        tc = time / @time_slice_cache_interval
-        if @before_tc == tc
-          key = @before_key
-        else
-          @before_tc = tc
-          key = @time_slicer.call(time)
-          @before_key = key
+        begin
+          tc = time / @time_slice_cache_interval
+          if @before_tc == tc
+            key = @before_key
+          else
+            @before_tc = tc
+            key = @time_slicer.call(time)
+            @before_key = key
+          end
+          formatted_data[key] ||= ''
+          formatted_data[key] << format(tag, time, record)
```

Comment:
`rescue` catches an exception happens at `format`. So, a TimeSlicedOutput plugin doesn't have to catch exception at `format`.
But `BufferedOutput#format_stream` doesn't catch exception. Thus a BufferedOutput plugin needs to catch exception at `format` to support error stream.
I think that is confusing. How about doing either of them?:

* a) TimeSliceOutput doesn't catch exceptions that happen at `format`
* b) BufferedOutput#format_stream catches exceptions that happen at `format` and calls `@router.emit_error_event`

I think a) is better at least with before v0.14 because BufferedOutput#format_stream can't decide message structure (like `{'record'=>record}` or `record`).